Don't add the hidden class name to elements with the hidden attribute (#3582)
woocommerce/woocommerce-gutenberg-products-block: docs/theming/README.md

In these cases, Blocks include some CSS resets to undo most default styles introduced by themes. A `<button>` that is displayed as a text link will probably have resets for the background, border, etc. That will solve most conflicts out-of-the-box but in some occasions those CSS resets might not have effect if the theme has a specific CSS selector with higher specificity. When that happens, we really encourage theme developers to decrease their selectors specificity so Blocks styles have preference, if that's not possible, themes can write CSS resets on top.

### Hidden elements

> WC Blocks use the [`hidden` HTML attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) in order to hide some elements from the UI so they are not displayed in screens neither read by assistive technologies. If your theme has some generic styles that tweak the CSS display property of a specific tag (ie: `div { display: block; }`). Make sure you account for the hidden case: `div[hidden] { display: none; }`.

Reviewer: ```suggestion
WC Blocks use the [`hidden` HTML attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) to hide some elements from the UI so they are not displayed in screens neither read by assistive technologies. If your theme has some generic styles that tweak the CSS display property of blocks elements (ie: `div { display: block; }`), make sure you correctly handle the hidden attribute: `div[hidden] { display: none; }`.
```

Tweak - started with the comma but also tweaked a few words here & there. Thanks for documenting this, happy with whatever you end up with 😄

Author: Much better now, thanks!
